### PR TITLE
Added CustomMenuTemplate.gs

### PR DESCRIPTION
**Pull request by ria-arora**

### CODE REVIEW

**Comment by ria-arora** [comment]
@googlebot I signed it!

File: CustomMenuTemplate.gs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+* Custom Menu with sub-menus
+* Adding an item to the editor's Add-on menu
+*
+* The onOpen event handler is triggered when the attached spreadsheet is opened.
+* Class SpreadsheetApp is used to access and create Google Sheets files.
+*/
+function onOpen() {
+  var ui = SpreadsheetApp.getUi(); 
+  // The getUi method returns an instance of the spreadsheet's UI environment.
+  // It allows the script to add features like menus, dialogs, sidebars.
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: Move this comment before the .getUi() statement.

File: CustomMenuTemplate.gs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  
+  //A simple Custom Menu
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: nit: space after //

File: CustomMenuTemplate.gs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      .addItem('Second item', 'menuItem2')
+      .addSeparator() 
+      .addItem('Third item', 'menuItem3')
+      .addToUi();  
+    
+  // Add an item to the editor's Add-on menu, under a sub-menu whose name is set automatically.
+  ui.createAddonMenu()
+      .addItem('First Item', 'menuItem1')
+      .addToUi();
+
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: nit: delete

File: CustomMenuTemplate.gs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+* Simple Custom Menu
+* Custom Menu with separators
+* Custom Menu with sub-menus
+* Adding an item to the editor's Add-on menu
+*
+* The onOpen event handler is triggered when the attached spreadsheet is opened.
+* Class SpreadsheetApp is used to access and create Google Sheets files.
+*/
+function onOpen() {
+  var ui = SpreadsheetApp.getUi();
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: rename ui variable to spreadsheetUi

File: CustomMenuTemplate.gs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+* Class SpreadsheetApp is used to access and create Google Sheets files.
+*/
+function onOpen() {
+  var ui = SpreadsheetApp.getUi(); 
+  // The getUi method returns an instance of the spreadsheet's UI environment.
+  // It allows the script to add features like menus, dialogs, sidebars.
+  
+  //A simple Custom Menu 
+  ui.createMenu('Custom Menu 1')
+      .addItem('First item', 'menuItem1') //Adds an item to the menu.
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: nit: space after //. here and everywhere else required.

File: CustomMenuTemplate.gs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+*/
+function menuItem3() {
+  var ui = SpreadsheetApp.getUi();
+  var response = ui.alert('Confirm', 'Are you sure you want to continue?', ui.ButtonSet.YES_NO);
+  // Process the user's response.
+  if (response == ui.Button.YES) {
+    ui.alert('The user clicked "Yes."');
+  } else {
+    ui.alert('The user clicked "No" or the close button in the dialog\'s title bar.');
+  }
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: Delete and just write a comment "// Add functionality to be performed on clicking the menu item here." Also, remove the comments above the function.

File: CustomMenuTemplate.gs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+     .alert('You clicked the first menu item!');
+}
+
+/**
+* Method alert(prompt, buttons) of Class Ui is used.
+* Opens a dialog box in the user's editor with the given message, and 'YES' and 'NO' set of buttons.
+* The user can also close the dialog by clicking the close button in its title bar.
+*/
+function menuItem2(){
+  var ui = SpreadsheetApp.getUi()
```
Comment (gargabhay) [inline CustomMenuTemplate.gs]: nit: semi colon. Did you test this script?

Comment (ria-arora) [reply CustomMenuTemplate.gs]: Yes, it worked fine then, probably due to automatic semicolon insertion. Made the correction.

File: custom_menu_template.gs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      .addToUi();  
+    
+  // Add an item to the editor's Add-on menu, under a sub-menu whose name is set automatically.
+  spreadsheetUi.createAddonMenu()
+      .addItem('First Item', 'menuItem1')
+      .addToUi();
+}
+
+function menuItem1() {
+  // Add functionality to be performed on clicking the menu item here.
```
Comment (gargabhay) [inline custom_menu_template.gs]: I meant only remove the implementation from menuItem3() function. You can keep menuItem1 and menuItem2 function implementations as before. Also, mark resolve conversation whenever you resolve a comment. :)

Comment (ria-arora) [reply custom_menu_template.gs]: Made the correction. 
Noted, I will resolve all of them from now!

**Review by gargabhay — APPROVED** [review]

**Review by paneendra-google — APPROVED** [review]

**Pull request merged**